tests/node.rs

```diff
@@ -0,0 +1,36 @@
+mod test;
+use test::prelude::*;
+
+#[test]
+fn at_least_one_node_should_be_available() {
+    let client = TestKubeClient::new();
+    let nodes = client.list_labeled::<Node>("kubernetes.io/arch=stackable-linux");
+
+    assert_that(&nodes.items).is_not_empty();
+}
+
+#[test]
+fn nodes_should_be_tainted() {
```

Comment:
If I delete all nodes then the agent creates a new node with the following taint:

```
effect: NoSchedule
key: node.kubernetes.io/not-ready
```

The node is running and could accept requests but this taint prevents pods from being scheduled on this node. I have to remove this taint manually. I do not know if this is the right thing to do.

However it is useful that the test fails when this taint is there, especially if it is the only node.

How should the test behave? Succeed if there is at least one node with the correct two taints and ignore all other nodes?

Reply:
I'm not sure there is a _right_ way for this test to behave, we probably need to agree on expectations that these tests can have for the cluster they are executed against.

Personally - but that is really just an opinion! - I think that each test should look at what it is supposed to look at in isolation. So in this case, the `nodes_should_be_tainted` would be happy if all our nodes that are present have the relevant taints, no matter if they are offline or online, since the check if one is online is part of `at_least_one_node_should_be_available`.

`cargo test service` works when the unreachable node is present, as I assume the scheduler would ignore that node when assigning the test pod.

Reply:
I agree. `nodes_should_be_tainted` checks if the Stackable taints are present but also allows now other taints. `at_least_one_node_should_be_available` requires now a node with only Stackable taints.